src/service.ts

```diff
@@ -145,15 +145,35 @@ export class Service {
       ...prevService.spec?.template?.metadata?.annotations,
       ...this.request.spec?.template?.metadata?.annotations,
     };
+    let name = this.request.spec?.template?.metadata?.name || '';
     this.request.spec!.template!.metadata = {
       annotations,
       labels,
     };
 
-    // Merge Revision Spec
+    if (!name) {
+      const prevRevisionName = prevService.spec!.template!.metadata!.name;
+      const revisionName = this.name;
+      let num;
+      if (prevRevisionName) {
+        const suffix = prevRevisionName!.split('-');
+        num = (parseInt(suffix[suffix.length - 2]) + 1).toString();
+      } else {
+        num = '1';
+      }
+      const newSuffix = `-${num.padStart(4, '0')}-${Math.random()
+        .toString(36)
+        .replace(/[^a-z]+/g, '')
+        .substring(0, 3)}`;
+      name = revisionName + newSuffix;
+    }
+
+    if (name.length > 63) name = name.substring(0, 63);
```

Comment:
IIUC we append a suffix to the name? Can we add some comments to this logic with string manipulations?

Reply:
Moved this logic to a separate method with better comments. When there is no perceived change in the service a new revision isn't being made. We can force the update with a suffix name change. This mimics the gcloud logic.